tensilelite/Tensile/Source/lib/include/Tensile/ContractionTaskPredicates.hpp

```diff
@@ -0,0 +1,131 @@
+                {
+                    const bool streamK = Debug::Instance().useExperimentalSelection() == 2;
+                    const bool streamKDP = Debug::Instance().useStreamKDataParrallel();
+
+                    if(task.problem.groupedGemm())
+                        return task.problem.workspaceSizeGroupedGemm() <= task.problem.workspaceSize();
+                    else if(streamK && streamKDP)
+                        return true;
+                    else
+                        return task.solution.requiredWorkspaceSize(task.problem, task.hardware) <= task.problem.workspaceSize();
+                }
+
+                virtual bool debugEval(Task const& task, std::ostream& stream) const override
+                {
+
+                    size_t gsu           = task.problem.getParams().gsu() > 0 ? task.problem.getParams().gsu()
+                                                                        : task.solution.sizeMapping.globalSplitU;
+                    size_t gsuMultiplier = gsu > 1 ? gsu : 0;
+                    const bool streamK = Debug::Instance().useExperimentalSelection() == 2;
+                    const bool streamKDP = Debug::Instance().useStreamKDataParrallel();
+
+                    if(task.problem.d().totalLogicalElements() * gsuMultiplier > MAX_GSU_WORKSPACE_SIZE)
+                        return debugEvalCmp(task,
+                                            stream,
+                                            "prob",
+                                            task.problem.d().totalLogicalElements() * gsuMultiplier,
+                                            "<=",
+                                            "max gsu workspace size",
+                                            MAX_GSU_WORKSPACE_SIZE);
+
+                    if(task.problem.groupedGemm())
+                        return debugEvalCmp(task,
+                                            stream,
+                                            "prob",
+                                            task.problem.workspaceSizeGroupedGemm(),
+                                            "<=",
+                                            "max",
+                                            task.problem.workspaceSize());
+                    if(streamK && streamKDP)
+                        return debugEvalCmp(task,
+                                            stream,
+                                            "prob",
+                                            0,
+                                            "<=",
+                                            "max",
+                                            task.problem.workspaceSize());
+
+                    return debugEvalCmp(task,
+                                        stream,
+                                        "prob",
+                                        task.solution.requiredWorkspaceSize(task.problem, task.hardware),
```

Comment:
Does requiredWorkspaceSize function detect streamKDP setting? It should return 0 workspace required for that case.

Reply:
good point - I've moved the logic from the predicate to inside requiredWorkspaceSize for a cleaner implementation